XamarinCommunityToolkit/Views/SegmentedView/iOS/SegmentedViewRenderer.ios.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using Xamarin.CommunityToolkit.iOS.UI.Views;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+using UIKit;
+using Xamarin.Forms.Platform.iOS;
+using System.Threading.Tasks;
+using System.Threading;
+using Foundation;
+using Xamarin.Forms.Internals;
+using Xamarin.CommunityToolkit.Extensions.iOS;
+
+[assembly: ExportRenderer(typeof(SegmentedView), typeof(SegmentedViewRenderer))]
+
+namespace Xamarin.CommunityToolkit.iOS.UI.Views
+{
+	public class SegmentedViewRenderer : ViewRenderer<SegmentedView, UISegmentedControl>
+	{
+		protected override void OnElementChanged(ElementChangedEventArgs<SegmentedView> e)
+		{
+			base.OnElementChanged(e);
+
+			if (e.NewElement == null)
+				return;
+
+			if (Control == null)
+				SetNativeControl(new UISegmentedControl());
+
+			if (e.OldElement != null && Control != null)
+				InvalidateControl();
+
+			if (e.NewElement != null)
+				InitializeControl();
+		}
+
+		void InitializeControl()
+		{
+			PopulateSegments(Element.Items);
```

Comment:
There can be a race condition on PopulateSegments since InsertSegment is async void and won't wait until all segments are inserted. My advise would be async all the way x)
so InitializeControl would be async, just like it would be in the android side.